source/loader/ur_ldrddi.cpp

```diff
@@ -4423,6 +4440,13 @@ __urdlllocal ur_result_t UR_APICALL urEventSetCallback(
 
     [[maybe_unused]] auto context = getContext();
 
+    // Replace the callback with a wrapper function that gives the callback the loader event rather than a
+    // backend-specific event
+    auto wrapper_data =
+        new event_callback_wrapper_data_t{pfnNotify, hEvent, pUserData};
```

Comment:
Yes, I know. Manual memory management.

But I'm not sure if there's a better C++-y way of doing this, does anyone have any ideas?

Reply:
I don't think so. It needs to survive a roundtrip to and from a bare pointer to be used in the API.

Reply:
yeah, i don't think we can rely on raii here, unless you'd like to manually play around with constructing/destroying unique pointers, but I don't see much point.